File: 2015/day4.py
import AOCInit
import util
import hashlib

inp = 'abcdef'

def computeMD5(s):
    return hashlib.md5(s.encode()).hexdigest()

# part 1
counter = 1
while True:
    if computeMD5(inp + str(counter))[:5] == '00000':
        break
    counter += 1
print(counter)

# part 2
while True:
    if computeMD5(inp + str(counter))[:6] == '000000':
        break
    counter += 1
print(counter)

File: 2015/day4.rb
require 'digest'

inp = File.read('input4').chomp

# part 1
p1 = (1..).find { |n| Digest::MD5.hexdigest(inp + n.to_s).match(/^0{5,}/) }
puts p1

# part 2
puts((p1..).find { |n| Digest::MD5.hexdigest(inp + n.to_s).match(/^0{6,}/) })
